### PR TITLE
SB 110335524 shared webpack config

### PR DESCRIPTION
**Pull request by sbrauer**
https://www.pivotaltracker.com/story/show/110335524

I think this has gotten to a point where I should open the first PR. It contains the webpack config that is common to [AG (on the `webpack` branch)](https://github.com/rentpath/ag/blob/webpack/webpack.config.js) and [m.ag (on the `dev` branch)](https://github.com/rentpath/m.apartmentguide/blob/dev/webpack.config.js). There will be later PRs to add base karma config, more docs (like migrating from Tautline), and of course app-specific PRs to use this repo in AG, etc.

My goal was to include as much config as possible that seemed generically useful.
For example, if using this repo AG's `webpack.config.js` could be shortened to just the loader and plugins that seemed app-specific to me:

``` javascript
var config = require('webpack-config-rentpath').config;

config.module.loaders.push({
  test: /marker-clusterer\.js$/,
  loader: 'exports?MarkerClusterer'
});

var webpack = require('webpack');
config.plugins.push(new webpack.ContextReplacementPlugin(/moment[\/\\]locale$/, /en/));
config.plugins.push(new webpack.ProvidePlugin({
  $: 'jquery',
  jQuery: 'jquery',
  Swiper: 'swiper',
  MarkerClusterer: 'marker-clusterer',
  strftime: 'strftime'
}));

module.exports = config;
```

I was initially going to do this as plain old Javascript, but switched to basing it on https://github.com/rentpath/generator-node-module after learning that all of our JS packages should use the generator. So there's some boilerplate that may not strictly be needed (in the first commit), and I took the opportunity to convert the base config to ES6. Not sure if that was the right move, since my impression is that `webpack.config.js` is normally not ES6. On the other hand, it might make maintenance more pleasant as we all become accustomed to ES6.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,87 @@
...
+config.module.loaders.push({ test: /everestjs\/index.js$/, loader: "script-loader" })
+
+// Add a plugin
+var webpack = require('webpack')
+config.plugins.push(new webpack.ContextReplacementPlugin(/moment[\/\\]locale$/, /en/))
+
+module.exports = config
+```
+
+If you need to make environment-specific config changes, you can take advantage of the `appEnv` variable that `webpack-config-rentpath` exports in addition to `config`. The value of `appEnv` is a string obtained from the environment variable `APPLICATION_ENVIRONMENT` and defaults to `"development"` if missing. For example:
```
Comment (c0) [inline README.md]: :+1:

**Comment by c0** [comment]
Sweet!

/cc @colinrymer @aribouius @mpelzsherman

**Comment by mpelzsherman** [comment]
Have you tested m.AG with this config, @sbrauer ?

**Comment by colinrymer** [comment]
We should default to production not development, but otherwise I dig it.

**Comment by colinrymer** [comment]
Explanation for defaulting to production: in case someone forgets to configure the app env, we don't want to accidentally ship code in development mode; granted this is JS, so it's not like folks can't dive into code if it's running on the client, but we want to make sure we make things as safe as possible by default (and webpack can be used for server side code).

**Comment by sbrauer** [comment]
@mpelzsherman: Not yet (there's a [separate story for it](https://www.pivotaltracker.com/story/show/110335648)), but I referred to the existing config a lot while deciding what should go into the shared config.
Here's my current TODO list for mdot's customizations:
- add everestjs loader
- update root.resolve with bowerDir and componentsDir
- add ResolverPlugin
- add ContextReplacementPlugin (moment locale)
- add the ProvidePlugin configured as before
- add commented SpritePlugin ???
- add .babelrc with { "presets": ['es2015', 'react'] }

**Comment by aribouius** [comment]
@colinrymer +1

Otherwise LGTM :+1:

**Comment by sbrauer** [comment]
Defaults to "production" now.

**Comment by colinrymer** [comment]
Is package this using commitizen & semantic-release?

**Comment by colinrymer** [comment]
Eh, retrolyzer isn't setup to work with commitizen formatted git messages yet, so nvm.

**Comment by Teapane** [comment]
I can add `commitizen` and `semantic-release`, but I suppose we should take a look at retrolyzer as well. @colinrymer.

LGTM :+1:

**Comment by c0** [comment]
Retrolyzer only picks up our main apps, right?

**Comment by c0** [comment]
To recap:
- Retrolyzer only looks at pivotal
- Wikigen looks at main apps, not libs to pull in story numbers
- Let's discuss `commitizen` and `semantic-release` outside this PR. It's a bigger change.

PR is :+1: from me.

**Comment by Teapane** [comment]
@c0 agreed. I can schedule something to chat about where we want to go with `commitizen` and `semantic-release`. 

:+1: :stars:

**Pull request merged**